library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/ApplicationServerSetsRelBrandDoctrineRepository.php

```diff
@@ -43,4 +44,42 @@ public function findByBrandId(int $brandId): array
 
         return $response;
     }
+
+    /**
+     * @return int[]
+     */
+    public function getApplicationServerSetIdsByBrandAdmin(AdministratorInterface $admin): array
+    {
+        if (!$admin->isBrandAdmin()) {
+            throw new \DomainException('User must be brand admin');
+        }
+
+        $qb = $this->createQueryBuilder('self');
+        $expression = $qb->expr();
+
+        $brandId = $admin->getBrand()?->getId();
+        if (is_null($brandId)) {
+            return [];
+        }
```

Comment:
I think this condition would be better outside, and send only the brand id as argument to this method.

Reply:
This is called directly from API. There is already some similar use cases, for example in CompanyRepository.getSupervisedCompanyIdsByAdmin()